File: JANUARY/06-01-2026/1161.cs
public class Solution {
    public int MaxLevelSum(TreeNode root) {
        Queue<TreeNode> q = new Queue<TreeNode>();
        q.Enqueue(root);
        int level = 1, ans = 1;
        long maxSum = long.MinValue;

        while (q.Count > 0) {
            int size = q.Count;
            long sum = 0;
            for (int i = 0; i < size; i++) {
                TreeNode node = q.Dequeue();
                sum += node.val;
                if (node.left != null) q.Enqueue(node.left);
                if (node.right != null) q.Enqueue(node.right);
            }
            if (sum > maxSum) {
                maxSum = sum;
                ans = level;
            }
            level++;
        }
        return ans;
    }
}

File: JANUARY/06-01-2026/1161.c
int maxLevelSum(struct TreeNode* root) {
    struct TreeNode* q[10001];
    int front = 0, rear = 0;
    q[rear++] = root;

    int level = 1, ans = 1;
    long long maxSum = -1000000000000000000LL;

    while (front < rear) {
        int size = rear - front;
        long long sum = 0;

        for (int i = 0; i < size; i++) {
            struct TreeNode* node = q[front++];
            sum += node->val;
            if (node->left) q[rear++] = node->left;
            if (node->right) q[rear++] = node->right;
        }

        if (sum > maxSum) {
            maxSum = sum;
            ans = level;
        }
        level++;
    }
    return ans;
}